File: Kernel/src/main/java/com/potato/kernel/Hardware/Sensor.java
package com.potato.kernel.Hardware;

import com.google.gson.annotations.SerializedName;

/**
 * fits the result from lhm, represent a sensor info it gives
 */
public class Sensor {
    @SerializedName("Id")
    private String id;

    @SerializedName("Name")
    private String name;

    @SerializedName("Index")
    private  int index;

    @SerializedName("Info")
    private String info;

    public String getInfo() {
        return info;
    }

    public int getIndex() {
        return index;
    }

    public String getName() {
        return name;
    }

    public String getId() {
        return id;
    }
}

File: ExternalTools/LibreHardwareMonitorWrapper/Lhm/Sensor.cs
using LibreHardwareMonitor.Hardware;

namespace LibreHardwareMonitorWrapper.Lhm;

public class Sensor : BaseHardware
{
    private readonly ISensor _mSensor;

    public Sensor(string id, string name, string info, ISensor sensor, int index) : base(id, name,
        info, index, HardwareType.Sensor)
    {
        _mSensor = sensor;
    }

    public int Value()
    {
        return _mSensor.Value.HasValue ? (int)_mSensor.Value : 0;
    }
}